### ISSUE TITLE
Add a configuration and setup guide for unit and integration testing in PHPStorm

### ISSUE THREAD

**Issue by mattiteraslahti** [markshust/docker-magento#181]
It would be awesome to have out-of-the-box configuration and setup guide for integration and unit testing in PHPStorm.

**Comment by markshust** [comment]
in progress

**Comment by dmanners** [comment]
Is there any update on this?
It would be really nice to have. I am happy to help out working on this.

**Comment by markshust** [comment]
@dmanners if you want to tackle this, that would be great! It will be a little while longer until I can get around to this.

**Comment by markshust** [comment]
@dmanners assigned this over to you, if you can't take this in, let me know and I'll add it to my queue so we can finally get this one in!

**Comment by jamesSaporito** [comment]
@markshust @dmanners Any update on this by any chance?

**Comment by dmanners** [comment]
@jamesSaporito finally got at least an IDEA setup guide sorted here. Will try to extract the customer specific change out and provide it back to the repo when I have time.

**Comment by dmanners** [comment]
Ok so this is the information we are using to run tests via the IDEA.

**Prerequisite** 

- Install the PHP Remote Interpreter plugin.

**Configure Docker Settings**

Under the section "Preferences | Build, Execution, Deployment | Docker" add the config as follows, making sure the local path matches your path on your machine.

![Docker-For-Mac-Config](https://user-images.githubusercontent.com/1572836/190664370-210fdc5e-a64d-4be0-b80c-c5203213c2da.png)

Next you will need a CLI interpreter and make sure you selected the right server and service:

![CliInterpretersIntegrationTest](https://user-images.githubusercontent.com/1572836/190664892-c16b3878-8c2c-4492-bed9-32556b7f4f33.png)

Next you need to setup the testframework, this needs to be done for all types so if you want to run Integration, Unit and Static tests you will need three frameworks here. This can be done under "Preferences | Languages & Frameworks | PHP | Test Frameworks". Make sure you have the right interpreter selected and that the path mapping is correct. The field "default config file" is where you can define if this config runs unit/integration/static test types.

![TestFrameWork](https://user-images.githubusercontent.com/1572836/190664801-de0b0f49-9e15-4897-afb8-a47230b2da8d.png)

Next make sure you have the correct interpreter selected for PHP. This can be found at "Preferences | Languages & Frameworks | PHP" and this can be using your generic interpreter and not your test specific ones.

![PhpSettings](https://user-images.githubusercontent.com/1572836/190665717-4f40e66e-7111-465c-9fbe-4f45ea87895d.png)

Once this is done you can create a PHPUnit config naming it appropriately for the test type, selecting the option "Defined in configuration file", selecting that test type's interpreter, and adding any runner config. 

![PHPUnitConfig](https://user-images.githubusercontent.com/1572836/190666234-56bd84b3-e06b-40d3-8666-683b8639783d.png)

For us we use the following runner options:

Testsuite | Testrunner options
-- | --
static | --testsuite "PHP Coding Standard Verification" --filter "testCodeStyle\|testCodeMess\|testStrictTypes\|testFixtureReuse"
integration | --debug --verbose
unit | --debug --verbose
unit-without-coverage | --debug --verbose

Once this is done in your IDEA you should see your PHPUnit config in the drop down, you select the one you wish to run and then hit the green play button.

![SelectToRun](https://user-images.githubusercontent.com/1572836/190666757-840a6e7b-b3ba-4fd6-965b-4443676d16cd.png)

It is also possible to run specific test classes and methods using these configs. You will need to open the file containing the test you wish to run. Right click on the green play button next to the test and select "modify config"

![Screenshot 2022-09-16 at 16 49 07](https://user-images.githubusercontent.com/1572836/190667455-1f6471c1-6194-47f8-85f8-57c75c4b1f33.png)

In the following pop-up select the correct interpreter, as well as if you wish to run a full class or single method.

![Screenshot 2022-09-16 at 16 50 50](https://user-images.githubusercontent.com/1572836/190667775-ccd68128-561a-45b1-9f82-47cc90c39502.png)

Once this is done the test config will be created and you can select this and run it using the drop down and green play button as all other test configs.

![Screenshot 2022-09-16 at 16 52 46](https://user-images.githubusercontent.com/1572836/190668118-80911296-9ce2-4d96-af68-36bd1c0fddd7.png)

Let me know if this works for you @jamesSaporito and if that does work I can try adding it to the main docs if that makes sense for you @markshust

**Comment by dmanners** [comment]
It might also be worth noting how we work this in a project.

We have a .dev folder in our project with all our testing required files like the phpunit.xml for all test types and mysql-config.php for integration tests. This is then mounted with our extension to the docker setup into the container so that we can use them in the above steps and not rely on the .dist versions.
